xc/xenops_server_xen.ml

```diff
@@ -2261,6 +2269,19 @@ module VGPU = struct
          Device.Dm.restore_vgpu task ~xs frontend_domid vgpu vcpus profile
       ) vm
 
+  let active_path vm vgpu = Printf.sprintf "/vm/%s/devices/vgpu/%s" vm (snd vgpu.Vgpu.id)
+
+  let set_active task vm vgpu active =
+    try
+      set_active_device (active_path vm vgpu) active
+    with e ->
+      debug "set_active %s.%s <- %b failed: %s" (fst vgpu.Vgpu.id) (snd vgpu.Vgpu.id) active (Printexc.to_string e)
```

Comment:
Do you need to reraise the exception here? Or can this fail due to race conditions (e.g. if the VM crashes during startup)

Reply:
I think we can raise an exception here, but I think VBD or VIF also need set_active_device in a startup which actually will not handle the exception at this point(set_active). 
For the race conditions,  could you explain more about the scenario? I think it may not fail.

Thanks.